compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -1525,19 +1525,22 @@ private void addTopLevelNode(BLangPackage pkgNode, TopLevelNode node) {
 
     private void defineErrorDetails(List<BLangTypeDefinition> typeDefNodes, SymbolEnv pkgEnv) {
         for (BLangTypeDefinition typeDef : typeDefNodes) {
-            if (typeDef.typeNode.getKind() != NodeKind.ERROR_TYPE) {
-                continue;
+            if (typeDef.typeNode.getKind() == NodeKind.ERROR_TYPE) {
+                BLangErrorType errorTypeNode = (BLangErrorType) typeDef.typeNode;
+                SymbolEnv typeDefEnv = SymbolEnv.createTypeEnv(errorTypeNode, typeDef.symbol.scope, pkgEnv);
+
+                BType detailType = Optional.ofNullable(errorTypeNode.detailType)
+                        .map(bLangType -> symResolver.resolveTypeNode(bLangType, typeDefEnv))
+                        .orElse(symTable.detailType);
+
+                ((BErrorType) typeDef.symbol.type).detailType = detailType;
+            } else if (typeDef.typeNode.type != null && typeDef.typeNode.type.tag == TypeTags.ERROR) {
+                BType detailType = ((BErrorType) typeDef.typeNode.type).detailType;
+                if (detailType == symTable.noType) {
+                    BErrorType type = (BErrorType) symResolver.resolveTypeNode(typeDef.typeNode, pkgEnv);
```

Comment:
Shouldn't we use a type env here also? If so can we refactor these two blocks to avoid repeating the same logic?

Reply:
Fixed the type-env, not quite sure how to fold those 2 branches